### PR TITLE
rest shifting according to clarified spec

### PR DESCRIPTION
**Pull request by mfwolffe**
This should take care of rests breaking the shifts. 

- rests will coalesce similar to how notes coalesce, e.g., if a terminating quarter rest is shifted once the resulting measure begins and ends with eighth rest. Another shift produces measure beginning with a quarter rest.
- doesn't seem to mind ... 'unusual'... motives (subjective I know), such as a measure with only one note

notes will now have an `isRest` property if they are part of a measure passed through these functions. Makes life easier.

### CODE REVIEW

File: components/flatEditor.js [hunk 1]
```diff
@@ -647,14 +647,18 @@ function FlatEditor({
                 setRefId(score.scoreId);
               })
               .catch((e) => {
-                e.message = `flat error: ${e.message}, not loaded from scoreId, score: ${JSON.stringify(score)}, orig: ${orig}, colors: ${colors}`;
-                if (debugMsg){
```
Comment (hcientist) [inline components/flatEditor.js:651]: ok, but like, can you still keep `debugMsg` when it's available at least in the console error. the point of the hasty change was to be able to distinguish which instances of the Flateditor are experiencing the errors?

**Review by hcientist — CHANGES_REQUESTED** [review]
Can you throw `debugMsg` somewhere when it's available?

File: components/flatEditor.js (hunk 1 above)
Comment (mfwolffe) [reply components/flatEditor.js:651]: Admittedly, my commenting out was hasty, especially considering the latter of the runtime errors seems to only happen when the scores lock. uncommented now

**Review by hcientist — APPROVED** [review]
ship it!

**Pull request merged**